subsys/net/lib/download_client/Kconfig

```
@@ -105,8 +105,8 @@ config DOWNLOAD_CLIENT_MAX_FILENAME_SIZE
 	range 8 256
 	default 192
 
-config DOWNLOAD_CLIENT_SOCK_TIMEOUT_MS
-	int "Receive timeout, in milliseconds"
+config DOWNLOAD_CLIENT_UDP_SOCK_TIMEO_MS
```

Should depend on COAP (https://github.com/nrfconnect/sdk-nrf/blob/827bd3e8cc4df9d1bdfae953ee386bd3a64bdb5b/subsys/net/lib/download_client/src/download_client.c#L617)

```suggestion
config DOWNLOAD_CLIENT_UDP_SOCK_TIMEO_MS
	depends on COAP
```

---

I had pushed a commit to address this, but this only forces me to `#ifdef` `socket_sectag_set` and drop `IS_ENABLED(COAP)` in `download_client_connect` in favor of another `#ifdef` there as well. I do not see any benefit of adding these `#ifdef` to hide a Kconfig from the user, which is stated explicitly to apply only for UDP sockets. It is better having that Kconfig visible and ignored when appropriate, and keep all the code compiled (instead of `#ifdef`-ed) so that we don't discover that it's broken only when we enable `COAP`, so I have reverted that change.

---

Regardless, I do not think it's worth blocking this PR because of that.

---

Agree, if it makes to code more complicated I believe the worse UX in the config menu is acceptable.